rules/S2589/metadata.json

```
@@ -3,7 +3,8 @@
   "type": "CODE_SMELL",
   "code": {
     "impacts": {
-      "MAINTAINABILITY": "MEDIUM"
+      "MAINTAINABILITY": "MEDIUM",
+      "SECURITY": "MEDIUM"
```

> Boolean expressions should not be gratuitous

Not covered by CFamily, so maybe revert these changes?
```suggestion
      "MAINTAINABILITY": "MEDIUM"
```